core/src/banking_stage.rs

```diff
@@ -802,9 +870,15 @@ impl BankingStage {
         unprocessed_packets: &mut UnprocessedPackets,
         packets: Packets,
         packet_indexes: Vec<usize>,
+        optional_transactions: Option<TransactionsAndOffsets>,
     ) {
         if !packet_indexes.is_empty() {
-            unprocessed_packets.push((packets, packet_indexes));
+            if unprocessed_packets.len()
+                > (Self::num_threads() * NUM_BATCHES_TO_BUFFER_PER_THREAD) as usize
```

Comment:
@pgarg66, @sagar-solana, is NUM_BATCHES_TO_BUFFER_PER_THREAD big enough here? If we start dropping packets, should we add a metric or is that not helpful in testnet?

Reply:
metrics will be useful..

Reply:
@sakridge, how big are each of these `packets` on average?

Reply:
Each entry is around ~250 packets in the case of something like bench-tps transactions. This will create about 400 batches per thread allowing all threads to buffer ~400,000 packets.

Reply:
I think something in the 500,000-1m range is reasonable.

Reply:
I will add metrics as well.